plugins/inputs/apcupsd/apcupsd_test.go

```diff
@@ -0,0 +1,129 @@
+package apcupsd
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"net"
+	"sync"
+	"testing"
+
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+func TestApcupsdGather(t *testing.T) {
+	input, ok := inputs.Inputs["apcupsd"]
+	if !ok {
+		t.Fatal("Input not defined")
+	}
+	apc := input().(*ApcUpsd)
+
+	apc.Description()
+	apc.SampleConfig()
+
+	one := boolToInt(true)
+	if one != 1 {
+		t.Errorf("boolToInt failed")
+	}
+
+	if testing.Short() {
+		t.Skip("Skipping network dependent tests in short mode")
+	}
+
+	ln, err := net.Listen("tcp", ":0")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ln.Close()
+
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		conn, err := ln.Accept()
+		if err != nil {
+			return
+		}
+
+		in := make([]byte, 128)
+		n, err := conn.Read(in)
+		if err != nil {
+			t.Fatal(fmt.Sprintf("failed to read from connection - %s", err.Error()))
```

Comment:
Use testify for these messages:
```
require.NoError(t, err)
```

If you want a custom error message:
```
require.NoError(t, err, "failed to read from connection")
```

Make sure you shutdown correctly...